docs/PYTHONIC_DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,67 @@
+---
+contact: { Eduard van Valkenburg }
+last updated: { 2024-08-06 }
+---
+
+# Pythonic Design Principles for Semantic Kernel
+
+## Context and Problem Statement
+
+The Pythonic design principles are a set of guidelines that help developers write code that is idiomatic to the Python programming language, while maintaining similar concepts as the dotnet version of Semantic Kernel. These principles are not strict rules, but rather a set of best practices that help developers write code that is easy to read, write, and maintain, and helps decide when and how to deviate from a design set forth for dotnet, as well as vice versa.
+
+While we try to maintain parity between dotnet, python and java for Semantic Kernel, it is also important to allow for language-specific idioms and for best practices to be followed. This decision will help guide the development of the Python version of Semantic Kernel.
+
+## Context
```

Comment:
Could you mention that Python has classmethod, static methods and instance methods, whereas dotnet has static methods / instance methods? I feel like it's worth calling out.